engine/src/dskmac.cpp

```diff
@@ -1601,6 +1601,11 @@ class MCStdioFileHandle: public MCSystemFileHandle
 	virtual bool Seek(int64_t offset, int p_dir)
 	{
         // TODO Add MCSystemFileHandle::SetStream(char *newptr) ?
+		if (m_is_eof)
```

Comment:
It is worth checking to see if this is a problem on any other platform.

It looks like android/iOS don't use a separate flag for eof - they just rely on feof (which presumably does play correctly with fseeko!).

Linux does have an `m_is_eof` flag but it appears to be unused... It uses feof (in IsExhausted) just as android/iOS do.

Windows, however, does appear to use m_is_eof in IsExhausted just as mac does. Its Seek method does not seem to reset it so the bug needs to be checked there and fixed in this PR if it exists.

(Confirming it isn't a problem on the other three platforms would be useful too!)

Reply:
Windows doesn't have this error - a successful read will unset m_is_eof. Theoretically IsExhausted could return true after a Seek but in practice IsExhausted is only ever checked after a call to Read

Reply:
@runrevmark confirmed this bug only affects macOS